encoding/thrift/inbound.go

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) 2016 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package thrift
+
+import (
+	"bytes"
+	"io/ioutil"
+
+	"github.com/yarpc/yarpc-go"
+	"github.com/yarpc/yarpc-go/transport"
+
+	"github.com/uber/thriftrw-go/protocol"
+	"github.com/uber/thriftrw-go/wire"
+	"golang.org/x/net/context"
+)
+
+// thriftHandler wraps a Thrift Handler into a transport.Handler
+type thriftHandler struct {
+	Handler  Handler
+	Protocol protocol.Protocol
+}
+
+func (t thriftHandler) Handle(ctx context.Context, treq *transport.Request) (*transport.Response, error) {
+	body, err := ioutil.ReadAll(treq.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	value, err := t.Protocol.Decode(bytes.NewReader(body), wire.TStruct)
+	if err != nil {
+		return nil, decodeError{Reason: err}
+	}
+
+	res, err := t.Handler.Handle(ctx, &Request{
+		Meta: yarpc.NewMeta(treq.Headers),
+		Body: value,
+	})
+
+	var headers map[string]string
+	if res.Meta != nil {
+		headers = res.Meta.Headers()
+	}
+
+	buffer := bytes.Buffer{}
```

Comment:
you use `var buffer bytes.Buffer` in the other file, but the struct constructor here -- I'm fine with either, but my preferences are either:
```go
var buf bytes.Buffer
```
or
```go
buf := &bytes.Buffer{}
```